Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
+			if (!m_StoreNetworkBackBuffer/* || g_UInputMan.KeyHeld(KEY_6)*/) {
+				g_SceneMan.Draw(pDrawScreen, pDrawScreenGUI, targetPos);
+			} else {
+				clear_to_color(pDrawScreen, g_MaskColor);
+				clear_to_color(pDrawScreenGUI, g_MaskColor);
+				g_SceneMan.Draw(pDrawScreen, pDrawScreenGUI, targetPos, true, true);
+			}
+			// Get only the scene-relative post effects that affect this player's screen
+			if (pActivity) {
+				g_PostProcessMan.GetPostScreenEffectsWrapped(targetPos, pDrawScreen->w, pDrawScreen->h, screenRelativeEffects, pActivity->GetTeamOfPlayer(pActivity->PlayerOfScreen(whichScreen)));
+				g_PostProcessMan.GetGlowAreasWrapped(targetPos, pDrawScreen->w, pDrawScreen->h, screenRelativeGlowBoxes);
 
-	} else {
-		Vector drawStartLeft;
-		Vector drawStartRight;
+				if (IsInMultiplayerMode()) { g_PostProcessMan.SetNetworkPostEffectsList(whichScreen, screenRelativeEffects); }
+			}
 
-		TranslateCoordinates(targetPos, m_Start, drawStartLeft, drawStartRight);
+			// TODO: Find out what keeps disabling the clipping on the draw bitmap
+			// Enable clipping on the draw bitmap
+			set_clip_state(pDrawScreen, 1);
 
-		pos[0] = drawStartLeft;
-		pos[1] = drawStartRight;
-		passes = 2;
-	}
+			//Always draw seam in debug mode
+#ifdef DEBUG_BUILD
+			g_PrimitiveMan.DrawLinePrimitive(Vector(0, 0), Vector(0, g_SceneMan.GetSceneHeight()), 5);
+#endif
 
-	Matrix rotation = Matrix(m_RotAngle);
-
-	for (int i = 0 ; i < 2 ; i++)
-	{
-		// Take into account the h-flipped pivot point
-		pivot_scaled_sprite(pDrawScreen,
-							m_pBitmap,
-							pos[i].GetFloorIntX(),
-							pos[i].GetFloorIntY(),
-							m_pBitmap->w / 2,
-							m_pBitmap->h / 2,
-							ftofix(rotation.GetAllegroAngle()),
-							ftofix(1.0));
-	}
-}
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          DrawBitmapPrimitive
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Schedule to draw a bitmap primitive.
-// Arguments:       Position of primitive in scene coordintaes, an entity to sraw sprite from, 
-//					rotation angle in radians, frame to draw
-// Return value:    None.
-
-	void FrameMan::DrawBitmapPrimitive(Vector start, Entity * pEntity, float rotAngle, int frame)
-	{
-		MOSprite * pMOS = dynamic_cast<MOSprite *>(pEntity);
-		if (pMOS)
-		{
-			BITMAP * pBitmap = pMOS->GetSpriteFrame(frame);
-			if (pBitmap)
-				m_Primitives.push_back(new BitmapPrimitive(start, pBitmap, rotAngle));
-		}
-	}
+			// Draw screen texts
+			int yTextPos = 0;
+
+			// Only draw this stuff for actual human players
+			if (whichScreen < g_ActivityMan.GetActivity()->GetHumanCount()) {
+				// Team of current screen's player
+				//int team = g_ActivityMan.GetActivity()->GetTeamOfPlayer(g_ActivityMan.GetActivity()->PlayerOfScreen(whichScreen));
+
+				yTextPos += 12;
+
+				// Message
+				if (!m_ScreenText[whichScreen].empty()) {
+					if (IsInMultiplayerMode()) {
```

Comment:
Running through diffchecker the only difference here is that when in multiplayer it uses `(GetPlayerFrameBufferWidth(whichScreen)` and `GetPlayerFrameBufferHeight(whichScreen)` instead of `GetPlayerScreenWidth()` and `GetPlayerScreenHeight()` in 3 places. Given that, clean up this duplicated code so it's just a few ternaries doing relevant stuff when in multiplayer mode.